### PR TITLE
feat: Add new fields and crud methods for Position

### LINKED ISSUE #258: [Backend] Add new fields and crud methods for Position/PositionDBConnector
🚨 **Attention: Please read the guidelines before applying; otherwise, you will be ignored.** 🚫📋
0. Please, join the telegram group to keep in touch https://t.me/+m0IGRiBvLtsyMzIy 
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. In [Position](https://github.com/djeck1432/spotnet/blob/main/web_app/db/models.py#L57) add new fields:
- is_protection_on, type boolean, default value is False
- liquidation_bonus, type float, default value is 0.0
- is_liquidated, type boolean, default value is False
- datetime_liquidation, type datetime, nullable=True
3. Add migrations file, how to do that, you can find [here](https://github.com/djeck1432/spotnet/blob/main/README.md#how-to-create-migration-file)
4. Add new crud method in [PositionDBConnector](https://github.com/djeck1432/spotnet/blob/main/web_app/db/crud.py#L233)
- `liduate_position` with parameters (position_id) will change value of `is_liquidated to True and will set current time datetime_liquidation
- `get_all_liqauted_position` -> will returl all position where `is_liquidated` is True

### PR DESCRIPTION
**Pull request by vestor-dev**
resolves #258

### CODE REVIEW

File: web_app/db/crud.py [hunk 1]
```diff
@@ -472,6 +472,71 @@ def save_current_price(self, position: Position, price_dict: dict) -> None:
...
+
+                self.write_to_db(position)
+                logger.info(f"Position {position_id} successfully liquidated.")
+                return True
+
+            except SQLAlchemyError as e:
+                logger.error(f"Error liquidating position {position_id}: {str(e)}")
+                db.rollback()
+                return False
+    def get_all_liquidated_positions(self) -> list[dict]:
```
Comment (djeck1432) [inline web_app/db/crud.py:504]: add space between methods

File: web_app/db/crud.py [hunk 2]
```diff
@@ -472,6 +472,71 @@ def save_current_price(self, position: Position, price_dict: dict) -> None:
...
+                liquidated_positions = (
+                    db.query(Position)
+                    .filter(Position.is_liquidated == True)
+                    .all()
+                )
+
+                # Convert ORM objects to dictionaries for return
+                return [
+                    {
+                        "id": position.id,
```
Comment (djeck1432) [inline web_app/db/crud.py:521]: Don't need `id`

File: web_app/db/crud.py [hunk 3]
```diff
@@ -472,6 +472,71 @@ def save_current_price(self, position: Position, price_dict: dict) -> None:
...
+        Retrieves all positions where `is_liquidated` is True.
+
+        :return: A list of dictionaries containing the liquidated positions.
+        """
+        with self.Session() as db:
+            try:
+                liquidated_positions = (
+                    db.query(Position)
+                    .filter(Position.is_liquidated == True)
+                    .all()
```
Comment (djeck1432) [inline web_app/db/crud.py:515]: Please, use `.scalar()`

File: web_app/db/crud.py [hunk 4]
```diff
@@ -472,6 +472,71 @@ def save_current_price(self, position: Position, price_dict: dict) -> None:
...
+                    {
+                        "id": position.id,
+                        "user_id": position.user_id,
+                        "token_symbol": position.token_symbol,
+                        "amount": position.amount,
+                        "multiplier": position.multiplier,
+                        "created_at": position.created_at,
+                        "status": position.status.value,
+                        "start_price": position.start_price,
+                        "is_protection": position.is_protection,
```
Comment (djeck1432) [inline web_app/db/crud.py:529]: don't need `is_protection` field

File: web_app/db/crud.py [hunk 5]
```diff
@@ -472,6 +472,71 @@ def save_current_price(self, position: Position, price_dict: dict) -> None:
...
+                        "liquidation_bonus": position.liquidation_bonus,
```
Comment (djeck1432) [inline web_app/db/crud.py:530]: don't need `liquidation_bonus` here, please remove it

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by vestor-dev** [comment]
Requested changes applied @djeck1432

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**